tools/recetox_aplcms/recetox_aplcms_recover_weaker_signals.xml

```diff
@@ -1,143 +1,91 @@
-<tool id="recetox_aplcms_recover_weaker_signals" name="RECETOX apLCMS - recover weaker signals" version="@TOOL_VERSION@+galaxy1">
+<tool id="recetox_aplcms_recover_weaker_signals" name="RECETOX apLCMS - recover weaker signals" version="@TOOL_VERSION@+galaxy0">
     <description>recover weaker signals from LC/MS spectra</description>
     <macros>
         <import>macros.xml</import>
-        <import>macros_split.xml</import>
     </macros>
     <expand macro="creator"/>
 
     <expand macro="requirements"/>
     <command detect_errors="aggressive"><![CDATA[
-        sh ${symlink_inputs} &&
-        Rscript -e 'source("${__tool_directory__}/utils.R")' -e 'source("${__tool_directory__}/main.R")' -e 'source("${run_script}")'
+        SAMPLE_NAME=\$(grep -Po '<run id="\K[^"]+' '$input_file') &&
+        Rscript -e 'source("${__tool_directory__}/utils.R")' -e 'source("${run_script}")'
     ]]></command>
     <configfiles>
-        <configfile name="symlink_inputs">
-            #for $infile in $ms_files
-                ln -s '${infile}' '${infile.element_identifier}'
-            #end for
-            #for $infile in $extracted_files
-                ln -s '${infile}' 'extracted_${infile.element_identifier}'
-            #end for
-            #for $infile in $corrected_files
-                ln -s '${infile}' '${infile.element_identifier}'
-            #end for
-        </configfile>
         <configfile name="run_script"><![CDATA[
-            #set filenames = str("', '").join([str($f.element_identifier) for $f in $ms_files])
-            filenames <- sort_samples_by_acquisition_number(c('$filenames'))
+            sample_name <- Sys.getenv("SAMPLE_NAME", unset = NA)
 
-            #set extracted_files = str("', '").join(["extracted_" + str($f.element_identifier) for $f in $extracted_files])
-            extracted <- load_features(c('$extracted_files'))
+            #set extracted_filenames = str("', '").join([str($f.element_identifier) for $f in $files])
+            extracted_feature_tables <- load_parquet_collection(extracted_filenames)
+            extracted_features <- select_table_with_sample_name(extracted_feature_tables, sample_name)
```

Comment:
This implementation has a problem. `load_parquet_collection` loads the parquet tables and converts them to matrices with `as.matrix`. After converting to matrices, the tables no longer store the attributes that the parquet files did. Hence, `select_table_with_sample_name` fails because sample names are no longer in the attributes. @xtrojak, is it necessary to convert the data frames to matrices?

Reply:
Besides that, `select_table_with_sample_name` works as expected with parquet data frames.